gcs/src/main/scala/quasar/blobstore/gcs/GCSListService.scala

```diff
@@ -101,19 +135,42 @@ object GCSListings {
       },{j => {
         val items = (j --\ "items").either
         val prefixes = (j --\ "prefixes").either
-        (items, prefixes) match {
-          case (Left(_), Left(_)) => DecodeResult.ok(GCSListings(List.empty[GCSFile]))
-          case (Left(_), Right(p)) => for {
+        val token = (j --\ "nextPageToken").either
```

Comment:
... then you can this here:
```suggestion
        val token = (j --\ "nextPageToken").as[Option[PageToken]]
```